src/components/plots/gridPlot.ts

```diff
@@ -48,23 +53,50 @@ export module Plot {
      */
     public project(attrToSet: string, accessor: any, scale?: Scale.AbstractScale<any, any>) {
       super.project(attrToSet, accessor, scale);
+
+      // Use x to determine x1 by default
+      if (attrToSet === "x") {
+        super.project("x1", (d: any, i: number, u: any, m: Plot.PlotMetadata) => {
+          return scale.scale(this._projections["x"].accessor(d, i, u, m));
+        });
+      }
+
+      // Use y to determine y1 by default
+      if (attrToSet === "y") {
+        super.project("y1", (d: any, i: number, u: any, m: Plot.PlotMetadata) => {
+          return scale.scale(this._projections["y"].accessor(d, i, u, m));
+        });
+      }
+
+      // Situation where x is defined but x2 is not defined
+      if (attrToSet === "x" && this._projections["x2"] === undefined) {
```

Comment:
The comment is sort of redundant, since what it says can be determined by examining the next line.